sycl/doc/extensions/proposed/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -218,9 +218,16 @@ methods on a modifiable graph. Each node represent either a command-group
 function, empty operation, or device memory allocation/free.
 
 | Edge
-| In the explicit graph building API edges are defined by the user. This is
-either through buffer accessors, the `make_edge()` function, or by passing
-dependent nodes on creation of a new node.
+| In the explicit graph building API edges are primarily defined by the user
+through newly added interfaces. This is either using the `make_edge()` function
+to define an edge between existing nodes, or using a
+`property::node::depends_on` property list when adding a new node to the graph.
+
+Edges can also be created when explicitly adding nodes to the graph through
+existing SYCL mechanisms for expressing dependencies. Data dependencies from
+buffer accessors to existing nodes in the graph are captured as an edge. Using
+`handler::depends_on()` will also create a graph edge when passed an event
+returned from a queue submission captured by a queue recording to the same graph.
```

Comment:
What happens if a node `depends_on` an event not coming from the recording in the graph? 

Does that create a dependency for the execution of the whole graph (in which case circular dependencies would arise) or is it simply ignored?

Reply:
Sorry, just now saw the text below. I assume that events coming from outside the recording would also be ignored here?

Would it be viable to have them create a dependency on the graph execution?

Reply:
> Sorry, just now saw the text below. I assume that events coming from outside the recording would also be ignored here?

Indeed, they'd be ignored here too. Will put some wording about that here so that readers don't have find it out later on indirectly.

> Would it be viable to have them create a dependency on the graph execution?

Yes, we recently merged this change https://github.com/reble/llvm/pull/26 for that purpose.

Reply:
>> Would it be viable to have them create a dependency on the graph execution?

> Yes, we recently merged this change https://github.com/reble/llvm/pull/26 for that purpose.

Ok, makes sense. Having the `depends_on` calls in the individual commands during recording create those dependencies might be even more convenient for the user. On the other hand, ignoring them and having separate `depends_on` on submission of the graph avoids a lot of problems with circular dependencies, so I think it's a good middle-ground.